__tests__/app.test.js

```diff
@@ -152,4 +153,58 @@ describe("app", () => {
         });
     });
   });
+  describe("POST /api/articles/:article_id/comments", () => {
+    test("POST 201: inserts a new comment to the db and returns the new comment back to the client", () => {
+      const newComment = {
+        username: "butter_bridge",
+        body: "Love this article!!!",
+      };
+      return request(app)
+        .post("/api/articles/2/comments")
+        .send(newComment)
+        .expect(201)
+        .then(({ body }) => {
+          expect(body.comments.author).toBe("butter_bridge");
+          expect(body.comments.body).toBe("Love this article!!!");
+        });
+    });
+    test("POST 400: returns bad request if any required field in the comments body is empty", () => {
+      const newComment = {
+        username: "butter_bridge",
+      };
+      return request(app)
+        .post("/api/articles/1/comments")
+        .send(newComment)
+        .expect(400)
+        .then(({ body }) => {
+          expect(body.msg).toBe("Bad Request");
+        });
+    });
+    test("GET 404: returns Not Found when given a non-existent article_id", () => {
+      const newComment = {
+        username: "butter_bridge",
+        body: "Love this article!!!"
+      };
+      return request(app)
+        .post("/api/articles/999/comments")
+        .send(newComment)
+        .expect(404)
+        .then(({ body }) => {
+          expect(body.msg).toBe("Invalid article_id");
+        });
+    });
+    test("GET 400: returns Bad Request when given an invalid article_id", () => {
+      const newComment = {
+        username: "butter_bridge",
+        body: "Love this article!!!"
+      };
+      return request(app)
+        .post("/api/articles/not-an-id/comments")
+        .send(newComment)
+        .expect(400)
+        .then(({ body }) => {
+          expect(body.msg).toBe("Bad Request");
+        });
+    });
+  });
```

Comment:
This testing is great! There is another test case that was missed however, you should include a test for when the username is not found 
_note: the username references the users table in the database, so psql will run a check to see if it exists and throw an error if it cannot be found_